server/expression.js

```diff
@@ -90,14 +91,15 @@ expression.query = {};
  * @param {*|{
  *   fileName: string
  * }} query
- * @param {string} expressionPath
+ * @param {string} dataPath
  * @return {expression.MatrixInfo|expression.Error}
  */
-expression.query.matrixInfo = function(query, expressionPath) {
+expression.query.matrixInfo = function(query, dataPath) {
   if (query.fileName === undefined) {
     return {error: 'fileName is undefined'};
   }
-  var file = expressionPath + query.fileName + '.data';
+  var expressionPath = dataPath + 'anonymous/' + expression.EXPRESSION_PREFIX_;
```

Comment:
username could be get via the user api
```js
user.getUsername();
```
There should be somewhere checking that the username retrieved is valid.